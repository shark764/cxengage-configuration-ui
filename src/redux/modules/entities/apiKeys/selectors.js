import { Map } from 'immutable';
import { getSelectedEntity } from '../selectors';
import { selectTenantRoles } from '../roles/selectors';

export const getInitialUpdateFormValues = state =>
  new Map({
    name: getSelectedEntity(state).get('name'),
    description: getSelectedEntity(state).get('description'),
    roleId: getSelectedEntity(state).get('roleId'),
    id: getSelectedEntity(state).get('id'),
    secret: getSelectedEntity(state).get('secret')
  });

export const getInitialCreateFormValues = state =>
  new Map({
    roleId: selectTenantRoles(state).length > 0 ? selectTenantRoles(state)[0].value : ''
  });
