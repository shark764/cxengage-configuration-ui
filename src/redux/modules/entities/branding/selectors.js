/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

export const getBranding = state =>
  state.getIn(['Entities', 'branding', 'data']);

export const getProtectedBranding = state =>
  state.getIn(['Entities', 'protectedBranding', 'data']);
