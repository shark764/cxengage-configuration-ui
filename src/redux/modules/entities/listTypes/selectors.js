/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import { createSelector } from 'reselect';

const getListTypes = state => state.getIn(['Entities', 'listTypes', 'data']);

export const getListTypesOptions = createSelector([getListTypes], listTypes =>
  listTypes
    .toJS()
    .filter(listType => listType.active)
    .map(listType => ({ value: listType.id, label: listType.name }))
);
