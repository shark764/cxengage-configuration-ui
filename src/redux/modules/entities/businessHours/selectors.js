import { selectFormInitialValues } from '../../form/selectors';
import { getSelectedEntity } from '../selectors';
import { Map } from 'immutable';

export const selectBusinessHoursFormInitialValues = state => {
  if (getSelectedEntity(state) === undefined) {
    const initialTimes = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun'].reduce(
      (r, key) => ({ ...r, [key.concat('StartTimeMinutes')]: 0, [key.concat('EndTimeMinutes')]: 0 }),
      {}
    );
    return new Map({ active: true, businessHoursType: '24/7', timezone: 'Etc/GMT+12', ...initialTimes });
  }
  return selectFormInitialValues(state);
};
