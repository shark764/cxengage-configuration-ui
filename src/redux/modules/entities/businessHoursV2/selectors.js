import { createSelector } from 'reselect';
import { Map, List } from 'immutable';

import { selectFormInitialValues } from '../../form/selectors';
import { getSelectedEntity } from '../selectors';

export const selectBusinessHoursV2FormInitialValues = state => {
  if (getSelectedEntity(state) === undefined) {
    return Map({ active: true, shared: false });
  }
  return selectFormInitialValues(state);
};

export const getBusinessHourV2Drafts = createSelector(
  getSelectedEntity,
  businessHour =>
    businessHour &&
    businessHour.get('drafts') &&
    List.isList(businessHour.get('drafts')) &&
    businessHour.get('drafts').toJS()
);
