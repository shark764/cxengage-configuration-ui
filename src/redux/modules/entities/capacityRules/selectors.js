import { createSelector } from 'reselect';
import { Map, List, fromJS } from 'immutable';

import { selectFormInitialValues } from '../../form/selectors';
import { getSelectedEntity, getSelectedSubEntity } from '../selectors';
import { ednRuleToImmutable } from './ednHelpers';

export const selectCapacityRulesFormInitialValues = (state) => {
  if (getSelectedEntity(state) === undefined) {
    return Map({ name: '' });
  }
  return selectFormInitialValues(state);
};

export const getCapacityRules = (state) => state.getIn(['Entities', 'capacityRules', 'data']);

export const getCapacityRulesSelector = createSelector([getCapacityRules], (capacityRules) =>
  capacityRules.filter((rule) => rule.get('active')).map((rule) => ({ label: rule.get('name'), value: rule.get('id') }))
);

export const selectCapacityRuleVersions = createSelector(
  getSelectedEntity,
  (capacityRule) =>
    capacityRule
      ? capacityRule
          .get('items', List([]))
          .map((capacityRule, i) => capacityRule.set('numericOrderVersion', `v${i + 1}`))
          .toJS()
      : List([]).toJS()
);

export const selectCapacityRuleVersionFormInitialValues = createSelector(
  getSelectedSubEntity,
  (version) =>
    version
      ? version.get('ruleSet') && version.set('rule', ednRuleToImmutable(version.get('ruleSet')))
      : fromJS({
          rule: {
            voice: true,
            groups: [
              {
                channels: ['email', 'messaging'],
                interactions: 4,
              },
            ],
          },
          quantifier: 'any',
        })
);
