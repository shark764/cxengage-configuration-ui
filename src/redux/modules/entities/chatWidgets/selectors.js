/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

import { Map, List } from 'immutable';
import { createSelector } from 'reselect';
import { getSelectedEntity } from '../selectors';
import { selectFormInitialValues, getCurrentFormValueByFieldName } from '../../form/selectors';

export const selectChatWidgetFormInitialValues = state => {
  if (getSelectedEntity(state) === undefined) {
    return new Map({
      brandColor: '#65758e',
      conversationColor: '#0099ff',
      actionColor: '#0099ff',
      displayStyle: 'button',
      buttonWidth: '58',
      buttonHeight: '58',
      prechatCapture: 'name'
    });
  }
  return selectFormInitialValues(state);
};

const getChatWidgets = state => state.getIn(['Entities', 'chatWidgets', 'data'], new List([]));
const getDigitalChannelsApps = state => state.getIn(['Entities', 'digitalChannelsApps', 'data'], new List([]));

export const getDigitalChannelsAppIds = createSelector(
  [getDigitalChannelsApps, getChatWidgets],
  (digitalChannelsApps, chatWidgets) =>
    digitalChannelsApps
      .filter(app => chatWidgets.filter(cw => cw.get('appId') === app.get('id')).size === 0)
      .map(app => ({ value: app.get('id'), label: app.get('name') }))
);

export const getDigitalChannelsApp = state => {
  return (
    getSelectedEntity(state) &&
    getSelectedEntity(state).get('appId') &&
    getDigitalChannelsApps(state).find(app => app.get('id') === getSelectedEntity(state).get('appId'))
  );
};

export const getDisplayStyleIsButton = state => getCurrentFormValueByFieldName(state, 'displayStyle') === 'button';
