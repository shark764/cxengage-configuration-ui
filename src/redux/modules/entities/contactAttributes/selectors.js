/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

import { createSelector } from 'reselect';
import { getEntities } from '../selectors';

export const getContactAttributes = createSelector(
  getEntities,
  entitites => entitites && entitites.getIn(['contactAttributes', 'data'])
);

export const getContactAttributesNames = createSelector(
  [getContactAttributes],
  contactAttributes =>
    contactAttributes && contactAttributes.size > 0
      ? contactAttributes
          .toJS()
          .map(attribute => attribute.objectName)
          .sort((a, b) => (a.toUpperCase() < b.toUpperCase() ? -1 : 0))
      : undefined
);

export const getMandatoryContactAttributes = createSelector(
  [getContactAttributes],
  contactAttributes => contactAttributes && contactAttributes.filter(attribute => attribute.get('mandatory') === true)
);
