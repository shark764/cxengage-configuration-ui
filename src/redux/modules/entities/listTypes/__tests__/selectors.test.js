/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import { fromJS } from 'immutable';
import { getListTypesOptions } from '../selectors';

describe('getListTypesOptions', () => {
  const state = fromJS({
    Entities: {
      listTypes: {
        data: [
          {
            id: '1',
            name: 'mock list type name 1',
            someOtherField: 'mock value',
            active: true
          },
          {
            id: '2',
            name: 'mock list type name 2',
            someOtherField: 'mock value',
            active: false
          }
        ]
      }
    }
  });
  it('converts the list types to javascript, filters out active false, and maps to the name and id', () => {
    expect(getListTypesOptions(state)).toMatchSnapshot();
  });
});
