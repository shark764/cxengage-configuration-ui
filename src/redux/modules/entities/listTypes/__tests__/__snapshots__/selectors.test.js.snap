// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getListTypesOptions list types defined converted to javascript, filters out active false, and maps to the name and id 1`] = `
Array [
  Object {
    "label": "mock list type name 1",
    "value": "1",
  },
]
`;
